### ISSUE TITLE
Midterm Feedback

### ISSUE THREAD

**Issue by kmn5409** [Pr3mR4jendran/CS506-Project---Analyzing-Lifestyle-and-Demographic-Risk-Factors-of-Diabetes-with-BRFSS-Data#7]
Currently for evaluation are you using a validation set? If possible it might be good to try and create a validation set and use that along with the test set to determine the hyperparameters for your models, as using the test set to determine hyperparameters is not the best practice. You can look at creating the validation set from the training set, and try to have a good representation of the classes. Additionally when testing your models it might be better to use scores like the F1 score since your dataset has an imbalance of the target classes since in your video you spoke about the F1 score mainly, I did see in the report you plan to use F1 for the future so just want to emphasize that it’s important.  

The plan of the advanced models seems good to me. I am a bit confused about the next steps in terms of visualizations. I understand the Confusion Matrices and ROC Curves still need to be done, but are there any other visualizations being done with the data? Do you plan to utilize SVD for your features into 2D space and determine any interesting patterns there if you color the points by classes. Additionally if you are using Decision Trees or XGBoost trying to look at the feature importance with SHAP plot. You can determine if there are other visualizations that might be interesting to show us for your report. Also it might be useful to try and include the visualizations in the Markdown so it makes it easier for us to see important graphs and so you think are important.
